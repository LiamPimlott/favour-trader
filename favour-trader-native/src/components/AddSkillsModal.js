import React, { Component } from 'react';
import { StyleSheet, View, Modal, KeyboardAvoidingView, Switch, Text } from 'react-native';
import { Card, Button, Icon, Input, ListItem } from 'react-native-elements';

class AddSKillsModal extends React.Component {
    constructor() {
        super();
        this.state = {
            isOpen: null,
            missingFields: false,
            addToHas: false,
            selectedSkill: -1,
        };
        this.renderAllSkills = this.renderAllSkills.bind(this);
    }

    toggleSelection(id) {
        if (this.state.selectedSkill._id === id) {
            this.setState({
                selectedSkill: -1,
            });
        } else {
            this.setState({
                selectedSkill: id,
            });
        }
    }

    renderAllSkills() {
        const { selectedSkill } = this.state;
        const toggle = () => this.toggleSelection;
        return (
            this.props.categories.map((item) => {
                const selected = selectedSkill === item._id;
                return (
                    <ListItem
                        containerStyle={{ backgroundColor: selected ? '#2089dc' : 'white'}}
                        hideChevron={true}
                        leftIcon={<Text>â‡Œ</Text>}
                        key={item._id}
                        title={item.skill}
                        accessible={true}
                        accessibilityLabel={`${item.skill}+ ' skill'`}
                        onPress={() => {
                            this.toggleSelection.call(this, item._id)
                        }}
                    />
                )
            })
        );
    }

    save = () => {
        if (this.selectedSkill !== -1) {
            const skillsType = this.state.addToHas ? 'has' : 'wants';
            this.props.onSubmit(skillsType, [{"category": this.state.selectedSkill}]);
        }
        this.props.toggle();
    }

    close = () => {
        this.props.toggle();
    }

    render() {
        return (
            <KeyboardAvoidingView behavior="padding">
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.isOpen}
                    onRequestClose={() => {
                        this.setState({ isOpen: false })
                    }}>
                    <View style={styles.textSwitchContainer}>
                        <Text style={styles.category}> {this.state.addToHas ? ('Add skills I have:') : ('Add skills I am looking for: ')}</Text>
                        <Switch value={this.state.addToHas} style={styles.switch}
                            onValueChange={() =>
                                this.setState({
                                    addToHas: !this.state.addToHas,
                                })} />
                    </View>
                    {
                        this.renderAllSkills()
                    }
                    <View style={styles.textSwitchContainer}>
                        <Button
                            title={"Save Skills"}
                            onPress={this.save}
                            buttonStyle={styles.submitButton}
                            accessible={true}
                            accessibilityLabel={'Save Skills'}
                        />
                        <Button
                            title={"Close"}
                            onPress={this.close}
                            buttonStyle={styles.closeButton}
                            accessible={true}
                            accessibilityLabel={'Close'}
                        />
                    </View>
                </Modal>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    textSwitchContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around'
    },
    category: {
        fontSize: 20,
    },
    switch: {
        transform: [{ scaleX: 1.0 }, { scaleY: 1.0}]
    },
    closeButton: {
        marginBottom: 15,
        marginTop: 15,
        width: 150,
        backgroundColor: 'red'
    },
    submitButton: {
        marginBottom: 15,
        marginTop: 15,
        width: 150,
    }
});

export default AddSKillsModal;