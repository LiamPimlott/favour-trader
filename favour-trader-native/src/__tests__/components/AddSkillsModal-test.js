import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import AddSkillsModal from '../../components/AddSkillsModal.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing AddSkillsModal component', () => {
    it('renders as expected', () => {
        const isOpen=false;
        const addSkill='';
        const categories={map:jest.fn()};
        const toggle='';
        const onSubmit='';
        const userProfile='';

        const wrapper = shallow(
            <AddSkillsModal isOpen={isOpen} addSkill={addSkill} categories={categories} toggle={toggle}
                            onSubmit={onSubmit} userProfile={userProfile}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});