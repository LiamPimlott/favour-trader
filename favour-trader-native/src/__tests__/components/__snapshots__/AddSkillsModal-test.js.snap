// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddSkillsModal component renders as expected 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={null}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
         
        Add skills I am looking for: 
      </Text>
      <Switch
        disabled={false}
        onValueChange={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
            ],
          }
        }
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <Button
        TouchableComponent={[Function]}
        buttonStyle={
          Object {
            "marginBottom": 15,
            "marginTop": 15,
            "width": 150,
          }
        }
        clear={false}
        disabled={false}
        iconRight={false}
        loadingProps={
          Object {
            "color": "white",
            "size": "small",
          }
        }
        onPress={[Function]}
        title="Save Skills"
      />
      <Button
        TouchableComponent={[Function]}
        buttonStyle={
          Object {
            "backgroundColor": "red",
            "marginBottom": 15,
            "marginTop": 15,
            "width": 150,
          }
        }
        clear={false}
        disabled={false}
        iconRight={false}
        loadingProps={
          Object {
            "color": "white",
            "size": "small",
          }
        }
        onPress={[Function]}
        title="Close"
      />
    </View>
  </Component>
</KeyboardAvoidingView>
`;
